src/ledgerwriter/src/test/java/anthos/samples/financedemo/ledgerwriter/TransactionValidatorTest.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2020, Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package anthos.samples.financedemo.ledgerwriter;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import static anthos.samples.financedemo.ledgerwriter.ExceptionMessages.
+        EXCEPTION_MESSAGE_INVALID_NUMBER;
+import static anthos.samples.financedemo.ledgerwriter.ExceptionMessages.
+        EXCEPTION_MESSAGE_NOT_AUTHENTICATED;
+import static anthos.samples.financedemo.ledgerwriter.ExceptionMessages.
+        EXCEPTION_MESSAGE_SEND_TO_SELF;
+import static anthos.samples.financedemo.ledgerwriter.ExceptionMessages.
+        EXCEPTION_MESSAGE_INVALID_AMOUNT;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+class TransactionValidatorTest {
+
+    private TransactionValidator transactionValidator;
+
+    @Mock
+    private Transaction transaction;
+
+    private static final String LOCAL_ROUTING_NUM = "123456789";
+    private static final String AUTHED_ACCOUNT_NUM = "1234567890";
+    private static final String NON_AUTHED_ACCOUNT_NUM = "0987654321";
+    private static final String TO_ACCOUNT_NUM = "5678901234";
+    private static final String TO_ROUTING_NUM = "567891234";
+    private static final Integer VALID_AMOUNT = 3755;
+
+    private static final String[] INVALID_ACCT_NUM = {
+        "12345678",
+        "123456789",
+        "12345678900",
+        "",
+        "12345678j",
+        "abcdefghij",
+        "123 456789",
+        "          ",
+        "1234567.89",
+        "123_456789",
+        "12345仮6789",
+        "12341545🐻"
+    };
+
+    private static final String[] INVALID_ROUTING_NUM = {
+        "12345678",
+        "1234567890",
+        "12345678900",
+        "",
+        "12345678j",
+        "abcdefghij",
+        "123 456789",
+        "          ",
+        "1234567.89",
+        "123_456789",
+        "12345仮6789",
+        "12341545🐻"
```

Comment:
🐻

Reply:
Wow it actually looks like a bear on my computer now! I feel like one of those people trying the color blind glasses for the first time :exploding_head: